torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -105,7 +105,8 @@ def inner(*flat_args):
 
         # It doesn't matter if we run this under predispatch or not because it is
         # only for figuring out metadata
-        with disable_above, FunctionalTensorMode():
+        mode = FunctionalTensorMode(_allow_token_discovery=False)
```

Comment:
I'm confused: this is the place we're collecting metadata. Shouldn't we be allowing discovery of new tokens?